game/gbtd_gbr0.ksy

```diff
@@ -0,0 +1,229 @@
+meta:
+  id: gbr0
+  endian: le
+  encoding: ascii
+  title: GBR file
+  license: MIT
+  application: GameBoy Tile Designer, GBTD
+  file-extension:
+    - gbr
+  tags:
+    - game
+
+doc: |
+  GBR0 files are used by GameBoy Tile Designer
+  to store tilemaps created with the program.
+doc-ref: https://www.devrs.com/gb/hmgd/gbtd.html
+
+seq:
+  - id: magic
+    contents: GBO0
+  - id: objects
+    type: gbr_object
+    repeat: eos
+
+types:
+  gbr_object:
+    seq:
+      - id: object_type
+        type: u2
+        enum: object_type
+      - id: object_id
+        type: u2
+      - id: record_length
+        type: u4
+      - id: body
+        size: record_length
+        type:
+          switch-on: object_type
+          cases:
+            object_type::producer:      object_producer
```

Comment:
BTW types can be nested. May make sense to nest them and remove the prefix.

Reply:
Ah nice, didn't know. Tried it.